### DISCUSSION TITLE
jvm/class-load

### DISCUSSION THREAD

**Discussion by giscus** [itwanger/tobebetterjavaer-giscus#222] [Announcements]
# jvm/class-load

Java程序员进阶之路，小白的零基础Java教程，从入门到进阶，我竟然不再抗拒Java的类加载机制了

https://tobebetterjavaer.com/jvm/class-load.html

**Comment by light-dot** [comment]
我很好奇，为啥双亲委派模型叫双亲委派模型，这个名字感觉也不贴切呀，感觉叫层层委派模型比较贴切。

**Reply by itwanger** [reply]
中文译名比较怪，科比布莱恩特应该叫布莱恩特，但大家都叫他科比

**Comment by chaofengdev** [comment]
`递归委派：这个过程会递归向上进行，从启动类加载器（Bootstrap ClassLoader）开始，再到扩展类加载器（Extension ClassLoader），最后到系统类加载器（System ClassLoader）。`
这里是不是写错了呀？递归向上不应该是向上吗？还是我理解错了呢。

**Reply by izilongwan** [reply]
确实有问题，下面的图才看明白

**Reply by fantasy-zrq** [reply]
这里优先使用ApplicationClassLoader，再是ExtClassLoader，最后才会走到BootStrapClassLoader，这个BootStrapClassLoader是最顶级的类加载器。

**Comment by Yolo-XYX** [comment]
你那个类加载器的图有问题吧，你把链接和初始化都放到加载器里，加载器只负责装载阶段呀

**Reply by 0319cc** [reply]
没问题啊，类加载器三个过程，loading，linking，init

**Comment by fortmin1** [comment]
这句话并不准确：“在准备阶段，类变量已经被赋过默认初始值，而在初始化阶段，类变量将被赋值为代码期望赋的值。”
实际上在准备阶段只是对static变量分配内存空间，如果是final static则会赋值。在初始化阶段，static变量会被赋值。另外感觉原句所在的段和下面的代码没什么联系。

**Reply by fortmin1** [reply]
1.自JDK8起Extension ClassLoader被替换为Plantform ClassLoader
2.双亲委派模型不是使用继承而是使用组合实现的
